client/extensions/wp-super-cache/preload-tab.jsx

```diff
@@ -30,28 +30,37 @@ import { isPreloadingCache } from './state/cache/selectors';
  */
 const renderCachePreloadInterval = ( {
 	handleChange,
-	isRequesting,
-	isSaving,
+	isDisabled,
 	preload_interval,
 } ) => (
 	<FormTextInput
 		className="wp-super-cache__preload-interval"
-		disabled={ isRequesting || isSaving }
+		disabled={ isDisabled }
 		min="0"
 		name="preload_interval"
 		onChange={ handleChange( 'preload_interval' ) }
 		step="1"
 		type="number"
-		value={ preload_interval || '' } />
+		value={ preload_interval || '0' } />
 );
 
 class PreloadTab extends Component {
-	state = {
-		preloadRefresh: true,
+	componentWillReceiveProps( nextProps ) {
+		this.state = {
+			preloadRefresh: parseInt( nextProps.fields.preload_interval, 10 ) === 0 ? false : true
```

Comment:
Don't we have to retain the initial state declaration (since `componentWillReceiveProps` isn't called on initial mount)?

Or, alternatively: Do we really need separate state and lifecycle methods here? Since `preloadRefresh` is directly derived from the `preload_interval`  field prop, we reflect that by adding a predicate method that returns `parseInt( this.props.fields.preload_interval, 10 ) !== 0` to use as `FieldToggle`'s `checked` and `renderCachePreloadInterval`'s `isDisabled` props.